### ISSUE TITLE
What doesn't work in vanilla Mycroft?

### ISSUE THREAD

**Issue by ChanceNCounter** [neatbasis/skill-fallback-chatgpt#1]
OVOS would sincerely appreciate reports in this regard, as we believe most of our skills to be backwards-compatible with Mycroft-core (except Dinkum.) However, they don't necessarily install clean through `msm`.

**Comment by neatbasis** [comment]
Depends on ovos-skills-manager which is not installed by default. Hangs on install with msm. Never heard of OVOS before discovering the OpenVoiceOS/skill-ovos-fallback-chatgpt. What's the recommended approach to installing on a vanilla Mycroft?

**Comment by ChanceNCounter** [comment]
I'm not surprised you haven't heard of us; we've been holding off on making announcements until MycroftAI's CEO follows up on Announcement Part I (which was now 6 weeks ago, so we're starting to feel foolish.)

You may or may not have spotted NeonAI, which is taking some of Mycroft's user support and MkII-related obligations. We are the open source project underneath Neon, KDE's Mycroft-based projects, and others. We all used to be prominent Mycroft community devs, and OVOS was just a superset of vanilla Mycroft, but when MycroftAI decided to create what is now the Mark II OS, we became a fork.

Tl;dr we haven't made the big announcement yet, but we are the surviving Mycroft project.

---

> What's the recommended approach to installing on a vanilla Mycroft?

I should clarify, when you say vanilla Mycroft, do you mean Mycroft-core cloned from GitHub, or do you mean a Mark II running its native software?

The native Mark II software is a fork MycroftAI created in 2022 when they realized they were never going to finish the project. It is unlikely to see continued development, and the long-term plan is to migrate as many users as possible to the less-polished but better-designed Neon system.

**Comment by JarbasAl** [comment]
> Depends on ovos-skills-manager which is not installed by default. Hangs on install with msm. Never heard of OVOS before discovering the OpenVoiceOS/skill-ovos-fallback-chatgpt. What's the recommended approach to installing on a vanilla Mycroft?

can you please open an issue in main repo, it is intended to run in old mycroft devices too not OVOS exclusive, if something is not working please let me know

what error did msm give you?

**Comment by neatbasis** [comment]
At the moment, there's no error indicated in msm install, but after installing here'sthe logs:
`2023-03-23 16:45:39.663 | INFO     | 17049 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: skill-ovos-fallback-chatgpt.openvoiceos
2023-03-23 16:45:39.680 - OVOS - ovos_utils.messagebus:add:442 - DEBUG - Added event: skill.converse.ping
2023-03-23 16:45:39.683 - OVOS - ovos_utils.messagebus:add:442 - DEBUG - Added event: skill.converse.request
2023-03-23 16:45:39.686 - OVOS - ovos_utils.messagebus:add:442 - DEBUG - Added event: skill-ovos-fallback-chatgpt.openvoiceos.activate
2023-03-23 16:45:39.690 - OVOS - ovos_utils.messagebus:add:442 - DEBUG - Added event: skill-ovos-fallback-chatgpt.openvoiceos.deactivate
2023-03-23 16:45:39.693 - OVOS - ovos_utils.messagebus:add:442 - DEBUG - Added event: intent.service.skills.deactivated
2023-03-23 16:45:39.696 - OVOS - ovos_utils.messagebus:add:442 - DEBUG - Added event: intent.service.skills.activated
2023-03-23 16:45:39.699 - OVOS - ovos_utils.messagebus:add:442 - DEBUG - Added event: mycroft.skill.enable_intent
2023-03-23 16:45:39.701 - OVOS - ovos_utils.messagebus:add:442 - DEBUG - Added event: mycroft.skill.disable_intent
2023-03-23 16:45:39.704 - OVOS - ovos_utils.messagebus:add:442 - DEBUG - Added event: mycroft.skill.set_cross_context
2023-03-23 16:45:39.706 - OVOS - ovos_utils.messagebus:add:442 - DEBUG - Added event: mycroft.skill.remove_cross_context
2023-03-23 16:45:39.709 - OVOS - ovos_utils.messagebus:add:442 - DEBUG - Added event: mycroft.skills.settings.changed
2023-03-23 16:45:39.712 - OVOS - ovos_utils.messagebus:add:442 - DEBUG - Added event: converse.skill.deactivated
2023-03-23 16:45:39.731 - OVOS - ovos_workshop.skills.base:load_dialog_files:1629 - DEBUG - No dialog loaded for en-us
2023-03-23 16:45:39.733 - OVOS - ovos_workshop.skills.base:load_vocab_files:1648 - DEBUG - No vocab loaded for en-us
2023-03-23 16:45:39.736 - OVOS - ovos_workshop.skills.base:settings:363 - ERROR - Skill not fully initialized. Only default values can be set, no settings can be read or changed.Move code from  __init__() to initialize() to correct this.
2023-03-23 16:45:39.738 - OVOS - ovos_workshop.skills.base:settings:363 - ERROR - Skill not fully initialized. Only default values can be set, no settings can be read or changed.Move code from  __init__() to initialize() to correct this.
2023-03-23 16:45:39.740 - OVOS - ovos_workshop.skills.fallback:remove_instance_handlers:228 - INFO - Removing all handlers...
2023-03-23 16:45:39.742 - OVOS - ovos_workshop.skills.base:settings:363 - ERROR - Skill not fully initialized. Only default values can be set, no settings can be read or changed.Move code from  __init__() to initialize() to correct this.
2023-03-23 16:45:39.745 | ERROR    | 17049 | mycroft.skills.skill_loader:_create_skill_instance:313 | Skill initialization failed with TypeError("__init__() got an unexpected keyword argument 'enable_cache'")
Traceback (most recent call last):
  File "/home/pi/mycroft-core/mycroft/skills/skill_loader.py", line 307, in _create_skill_instance
    self.instance.initialize()
  File "/opt/mycroft/skills/skill-ovos-fallback-chatgpt.openvoiceos/__init__.py", line 35, in initialize
    self._chat = OpenAIPersonaSolver(config=self.settings)
  File "/home/pi/mycroft-core/.venv/lib/python3.7/site-packages/ovos_solver_openai_persona/__init__.py", line 9, in __init__
    super().__init__(name="OpenAI ChatGPT Persona", config=config)
  File "/home/pi/mycroft-core/.venv/lib/python3.7/site-packages/ovos_solver_openai_persona/engines.py", line 58, in __init__
    enable_cache=False, enable_tx=False)
TypeError: __init__() got an unexpected keyword argument 'enable_cache'
2023-03-23 16:45:39.748 | ERROR    | 17049 | mycroft.skills.skill_loader:_communicate_load_status:351 | Skill skill-ovos-fallback-chatgpt.openvoiceos failed to load
`

**Comment by neatbasis** [comment]
By vanilla I mean Mycroft-core cloned from GitHub. Specifically Picroft disk image by mycroft-ai.

**Comment by neatbasis** [comment]
Well I did have to update it a bit to keep it alive. Glad to hear that the project lives on with OVOS and NeonAI. Looks like I'll be migrating in the near future. Thank you for the info.
